pkg/reconciler/brokercell/controller.go

```diff
@@ -67,7 +71,27 @@ func NewController(
 
 	brokercellInformer.Informer().AddEventHandlerWithResyncPeriod(controller.HandleAll(impl.Enqueue), reconciler.DefaultResyncPeriod)
 
-	// TODO: add additional informer event handlers here.
+	// Watch data plane components created by brokercell so we can update brokercell status immediately.
+	// 1. Watch deployments for ingress, fanout and retry
+	deploymentinformer.Get(ctx).Informer().AddEventHandler(handleResourceUpdate(impl))
+	// 2. Watch ingress endpoints
+	endpointsinformer.Get(ctx).Informer().AddEventHandler(handleResourceUpdate(impl))
+	// 3. Watch hpa for ingress, fanout and retry deployments
+	hpainformer.Get(ctx).Informer().AddEventHandler(handleResourceUpdate(impl))
 
 	return impl
 }
+
+// handleResourceUpdate returns an event handler for resources created by brokercell such as the ingress deployment.
+func handleResourceUpdate(impl *controller.Impl) cache.ResourceEventHandler {
+	return controller.HandleAll(func(obj interface{}) {
+		if mo, ok := obj.(metav1.Object); ok {
+			bcName, exist := mo.GetLabels()[resources.BrokerCellLabelKey]
+			if !exist {
+				// This object is not created by brokercell
+				return
+			}
+			impl.EnqueueKey(types.NamespacedName{Namespace: mo.GetNamespace(), Name: bcName})
+		}
+	})
+}
```

Comment:
Can you use something like this? Since they're "children" of the brokercell.

```
cache.FilteringResourceEventHandler{
  FilterFunc: controller.FilterGroupKind(appsv1.Kind("Deployment")),
  Handler:    controller.HandleAll(impl.EnqueueControllerOf),
}
```

Reply:
Unfortunately this doesn't work for endpoints (the endpoints are indirectly created by the ingress service and therefore doesn't have the owner reference). I could do this for other resources if you prefer, but the current approach is generic for all resource created by brokercell.

Reply:
But the annotations will be passed to the endpoint? Interesting. Maybe add a comment explaining why of doing this.

Reply:
In fact, I found an existing helper function that does similar to what I do. Changed to that :)

I suppose we don't need a `FilterFunc` here. The filter func is applied to the object in the informer, for instance, in the hpaInformer. I think the objects we get from the hpsInformer should be an HPA object.

Reply:
Nice. Good to know.